velox/dwio/common/Common.cpp

```diff
@@ -15,11 +15,47 @@
  */
 
 #include "velox/dwio/common/Common.h"
+#include "velox/common/base/Exceptions.h"
 
 #include <folly/Conv.h>
 
 namespace facebook::velox::dwio::common {
 
+std::unique_ptr<folly::io::Codec> compressionKindToCodec(CompressionKind kind) {
+  switch (kind) {
+    case CompressionKind_NONE:
+      return getCodec(folly::io::CodecType::NO_COMPRESSION);
+    case CompressionKind_ZLIB:
+      return getCodec(folly::io::CodecType::ZLIB);
+    case CompressionKind_SNAPPY:
+      return getCodec(folly::io::CodecType::SNAPPY);
+    case CompressionKind_ZSTD:
+      return getCodec(folly::io::CodecType::ZSTD);
+    case CompressionKind_LZ4:
+      return getCodec(folly::io::CodecType::LZ4);
+    default:
+      VELOX_UNSUPPORTED(
+          "Not support {} in folly", compressionKindToString(kind));
+  }
+}
+
+CompressionKind codecTypeToCompressionKind(folly::io::CodecType type) {
+  switch (type) {
+    case folly::io::CodecType::NO_COMPRESSION:
+      return CompressionKind_NONE;
+    case folly::io::CodecType::ZLIB:
+      return CompressionKind_ZLIB;
+    case folly::io::CodecType::SNAPPY:
+      return CompressionKind_SNAPPY;
+    case folly::io::CodecType::ZSTD:
+      return CompressionKind_ZSTD;
+    case folly::io::CodecType::LZ4:
+      return CompressionKind_LZ4;
+    default:
+      VELOX_UNSUPPORTED("Not support folly codec type {}", type);
```

Comment:
If I change to this one, I will get error:
```
this constant expression has type "folly::io::CodecType" instead of the required "int" type
```
Wonder why not switch enum type?

Reply:
Maybe because `enum class CodecType`